autofetch/views.py

```diff
@@ -0,0 +1,137 @@
+
+    print(f"Start: {str(datetime.now())} \n")
+
+    for city_name in city_list:
+
+        print(f"Fetching {city_name}............")
+
+        try:
+            results = fetch_craigslist_housing(
+                limit=limit, site="newyork", category="apa", area=city_name
+            )
+
+        except AttributeError:
+            print(
+                f" Cannot write into city: {city_name}, because one of the url not exist\n"
+            )
+            continue
+
+        for r in results:
+
+            if len(Apartment.objects.filter(c_id=r["id"])) > 0:
+                continue
+
+            # check if is_existed w/ address reversed from lat,lon
+            if r["geotag"] is not None:
+
+                address, city, state, zipcode, full_address = "", "", "NY", 11201, ""
+
+                lat = r["geotag"][0]
+                lon = r["geotag"][1]
+
+                try:
+                    reverse_response = fetch_reverse_geocode((lat, lon))
+
+                except googlemaps.exceptions.TransportError:
+                    print(googlemaps.exceptions.TransportError)
+                    continue
+
+                if "formatted_address" in reverse_response[0].keys():
+                    full_address = reverse_response[0]["formatted_address"]
+
+                normalize_addr_dic = normalize_us_address(full_address)
+
+                if not normalize_addr_dic:
+                    print("return None from nomalize_us_address")
+                    print(f"The address input is: {full_address}")
+                    continue
+
+                state = normalize_addr_dic.state
+                address = normalize_addr_dic.street
+                city = normalize_addr_dic.city
+                zipcode = normalize_addr_dic.zipcode
+
+            else:
+                print("No geotag in craiglist results")
+                continue
+
+            # match the not null constraint of postgre
+            if state is None or address is None or city is None or zipcode is None:
+                print("state, address, city or zipcode is None")
+                continue
+
+            loc, loc_created = Location.objects.get_or_create(
+                address=address, city=city, state=state, zipcode=zipcode
+            )
+
+            if loc_created:
+                loc.latitude = lat
+                loc.longitude = lon
+                loc.save()
+
+            c_id = r["id"]
+            price = int(r["price"].split("$")[1])
+            last_updated = r["last_updated"]
+            bedrooms = r["bedrooms"]
+
+            apartment, apa_created = Apartment.objects.get_or_create(
+                c_id=c_id, location=loc
+            )
+
+            if apa_created:
+                apartment.rent_price = price
+                apartment.number_of_bed = bedrooms
+                apartment.last_modified = last_updated
+                apartment.save()
+            elif apartment.last_modified != last_updated:
```

Comment:
Will this point ever be reached during execution? It seems to me that the line `if len(Apartment.objects.filter(c_id=r["id"])) > 0: continue` will insure that any processing will be skipped if the apartment already exists in our db

Reply:
@ab7289 Good catch! Modified.